os/arch/arm/src/armv8-m/up_blocktask.c

```diff
@@ -189,6 +189,12 @@ void up_block_task(struct tcb_s *tcb, tstate_t task_state)
 #endif
 #ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
 				up_mpu_set_register(rtcb->stack_mpu_regs);
+#endif
+#ifdef CONFIG_REG_STACK_OVERFLOW_PROTECTION
```

Comment:
minor one.
To say these are the same functionality, how about using below?
```
#if defined(CONFIG_MPU_xxxx)

#elif defined(CONFIG_REG_xxx)

#endif 
```

Reply:
changed